app/src/main/java/com/vero/tasky/agenda/data/worker/CreateEventWorker.kt

```diff
@@ -0,0 +1,71 @@
+package com.vero.tasky.agenda.data.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.squareup.moshi.Moshi
+import com.vero.tasky.agenda.data.local.dao.EventDao
+import com.vero.tasky.agenda.data.local.dao.ModifiedAgendaItemDao
+import com.vero.tasky.agenda.data.local.entities.ModifiedAgendaItemEntity
+import com.vero.tasky.agenda.data.mappers.toCreateEventRequest
+import com.vero.tasky.agenda.data.mappers.toLocalPhoto
+import com.vero.tasky.agenda.data.remote.network.api.EventApi
+import com.vero.tasky.agenda.data.remote.network.request.CreateEventRequest
+import com.vero.tasky.agenda.data.util.multipart.MultipartParser
+import com.vero.tasky.agenda.domain.model.AgendaItemType
+import com.vero.tasky.agenda.domain.model.ModificationType
+import com.vero.tasky.agenda.domain.workmanagerrunner.UpdateEventWorkerRunner
+import com.vero.tasky.core.data.remote.safeSuspendCall
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import okhttp3.MultipartBody
+
+@HiltWorker
+class CreateEventWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val multipartParser: MultipartParser,
+    private val modifiedAgendaItemDao: ModifiedAgendaItemDao,
+    private val eventDao: EventDao,
+    private val api: EventApi,
+    private val moshi: Moshi,
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
+        val eventId = inputData.getString(UpdateEventWorkerRunner.EVENT_ID) ?: ""
```

Comment:
If eventid is null, you could just return Result.success or something

No need to make it empty and continue running the worker